adapter/internal/operator/controllers/dp/api_controller.go

```diff
@@ -282,6 +300,9 @@ func (apiReconciler *APIReconciler) resolveAPIRefs(ctx context.Context, prodHTTP
 		}
 		prodHTTPRoute.Authentications = authentications
 		prodHTTPRoute.RateLimitPolicies = rateLimitPolicies
+		prodHTTPRoute.ResourceAuthentications = resourceAuthentications
+		prodHTTPRoute.ResourceRateLimitPolicies = resourceRateLimitPolicies
+		prodHTTPRoute.ResourceAPIPolicies = resourceAPIPolicies
```

Comment:
`ResourceAPIPolicies` should be passed inside the resolveHttpRouteRefs method and set up before calling `apiReconciler.getResolvedBackendsMapping(ctx, httpRouteState)` because that method resolves the backend mappings for API policies and resource policies. There was a bug related to the backendRef resolution in apipolicies and I fixed it with #1006 .

Reply:
Yes, to fix that we can move following LOC to here from `resolveHttpRouteRefs`
```
httpRouteState.BackendMapping = apiReconciler.getResolvedBackendsMapping(ctx, httpRouteState)
```